### DISCUSSION TITLE
Bitcoind wallet state

### DISCUSSION THREAD

**Discussion by kiwiidb** [ACINQ/eclair#2583] [Q&A] [answered]
Does Eclair store state on bitcoind for the wallet? If you would start over with a completely new bitcoind do you need to do anything to recover the funds on the bitcoind side?

**Comment by t-bast** [comment, marked as answer]
Yes, eclair uses bitcoind to manage the non-lightning on-chain wallet part, as described here: https://github.com/ACINQ/eclair#prerequisite-bitcoin-core

So you'll need to backup your bitcoind wallet independently of eclair's seeds.

You should be able to switch to another bitcoind wallet somewhat transparently: but the channels that were previously opened will close to an on-chain address handled by the previous bitcoind wallet. @sstone is working on fixing that and will publish detailed guides on how to migrate your bitcoind wallet while preserving channels.
